x/programs/examples/counter-external/src/lib.rs

```diff
@@ -14,3 +14,22 @@ pub fn get_value(ctx: &mut Context, external: Address, of: Address) -> u64 {
     let ctx = ctx.to_extern(external, 1_000_000, 0);
     counter::get_value(&ctx, of)
 }
+
+#[cfg(test)]
+mod tests {
+    use crate::*;
+
+    #[test]
+    fn get_value_exists() {
+        let mut ctx = Context::new();
+        let external = Address::new([0; 33]);
+        let of = Address::new([1; 33]);
+
+        // value `of` has in the external contract
+        let value = 5_u64;
+        ctx.mock_call_function(external, "get_value", of, 1_000_000, 0, value);
```

Comment:
not the greatest fan of this syntax but I think its fine for now. 

Also, we could remove the `max_gas` and `value` params to simplify